manifest: update openthread feature sets and docs (#6589)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

------

* Changed how Thread 1.2 is enabled, refer to :ref:`thread_ug_thread_specification_options` and :ref:`thread_ug_feature_sets` for more information.

> * Modified feature sets with support for SRP client and removal or rarely used features.

Reviewer: ```suggestion
* Modified feature sets with support for SRP client and removal of rarely used features.
```

> * Modified feature sets with support for SRP client and removal of rarely used features.

Reviewer 2: ```suggestion
* Modified feature sets by adding support for SRP client and removing rarely used features.
```
